### ISSUE TITLE
Ivy: Animation events removed when node's position is changed

### ISSUE THREAD

**Issue by ghostlytalamaur** [angular/angular#37895]
# 🐞 bug report

### Affected Package

 The issue is caused by package @angular/animations

### Is this a regression?

 Yes, without ivy animation events don't removed.

### Description

Animation events is removed from `AnimationTransitionNamespace` during change node's position inside `ViewContainerRef`. This causes issue inside `mat-tab-group` when tab's order is changed because displaying content depends on animation events.
When node's index is changed a node detaches from old position and inserts into new one. During detach animation events are removed, but not added when node inserts back into new position. 

## 🔬 Minimal Reproduction

[StackBlitz](https://stackblitz.com/github/ghostlytalamaur/angular-rearrange-tab-bug) [GitHub Repo](https://github.com/ghostlytalamaur/angular-rearrange-tab-bug)
#### Steps to reproduce:
1. Open StackBlitz
2. Enable Ivy in settings
3. Click on rearrange button
4. Click on Tab 2. There is no content.

**Comment by tenglongroy** [comment]
also another reproduction here.
https://stackblitz.com/edit/angular-ivy-ehgtbm

**Comment by Dalomo** [comment]
This has not been resolved for more than a year now. Is there any update?
For now, I recommend anyone facing this problem to turn ivy off until it is resolved.

````json
  "angularCompilerOptions": {
    "enableIvy": false
  }
````

**Comment by Bschitter** [comment]
I'm also having the same issue with mat-tab-group on Angular 13.1.1 and with Ivy.
Really don't want to turn off Ivy...

**Comment by Bettaswamy** [comment]
Any updates on this issue? Still facing the same issue with mat-tab-group on Angular 13.1.1

**Comment by Bschitter** [comment]
> > I'm also having the same issue with mat-tab-group on Angular 13.1.1 and with Ivy. Really don't want to turn off Ivy... Edit: As I see, from version 12 you cannot turn it off anymore.
> 
> @Bschitter did you find any workaround for this?

@Bettaswamy No, unfortunately not. I ended up changing my application such that the tabs are not reordered anymore.

**Comment by DustinPham** [comment]
@Bettaswamy @Bschitter I found a working example of the mat-tab reordering here if you both still needed it: https://stackblitz.com/edit/angular-ivy-bvhnt7?file=src%2Fapp%2Fapp.component.ts

found from this git issue: https://github.com/angular/components/issues/13572

**Comment by dalokey** [comment]
Thank you for sharing the solution!

**Comment by Bschitter** [comment]
@DustinPham thanks! Although I don't need it anymore.

**Comment by ojpbay** [comment]
The solution from @DustinPham works because of the trackBy function. If you remove that (or track by something other than index) the issue with tab content being lost returns.

**Comment by thePunderWoman** [comment]
This issue was closed as obsolete due to the introduction of the new `animate.enter` and `animate.leave` APIs. The `@angular/animations` package is now deprecated and will be removed in Angular v23. We hope you give the new APIs a try.
